### PR TITLE
Send messages on events where the marathon appId matches given regex

### PR DESCRIPTION
**Pull request by eodgooch**
In a large data-center the operations team would like to be notified of marathon events matching some Regular Expression pattern. This pull request allows users to specify a Marathon env var `APP_ID_REGEXS` which is a comma separated list of strings to filter events by.

### CODE REVIEW

**Comment by tobilg** [comment]
Thanks for the PR! Could you please add an example of the usage to the Readme as well?

File: index.js [hunk 1]
```diff
@@ -33,16 +34,53 @@ if (process.env.EVENT_TYPES) {
     options.eventTypes = ["deployment_info", "deployment_success", "deployment_failed", "deployment_step_success", "deployment_step_failure", "group_change_success", "group_change_failed", "failed_health_check_event", "health_status_changed_event", "unhealthy_task_kill_event"]
 }
 
+if (process.env.APP_ID_REGEXS) {
+    // Use environment variable
+    if (process.env.APP_ID_REGEXS.indexOf(",") > -1) {
```
Comment (tobilg) [inline index.js]: Is splitting by `,` for regexes? Will only work it the regexes themselves don't contain a `,`

File: index.js [hunk 2]
```diff
@@ -33,16 +34,53 @@ if (process.env.EVENT_TYPES) {
     options.eventTypes = ["deployment_info", "deployment_success", "deployment_failed", "deployment_step_success", "deployment_step_failure", "group_change_success", "group_change_failed", "failed_health_check_event", "health_status_changed_event", "unhealthy_task_kill_event"]
 }
 
+if (process.env.APP_ID_REGEXS) {
```
Comment (tobilg) [inline index.js]: I'd suggest to call the env var `APP_ID_REGEXES`, see https://en.wiktionary.org/wiki/regex#Noun

File: index.js [hunk 3]
```diff
@@ -33,16 +34,53 @@ if (process.env.EVENT_TYPES) {
...
+}
+
 // Placeholder for the handler functions
 let handlers = {};
 
 // Populate handler functions
 options.eventTypes.forEach(function (eventType) {
     handlers[eventType] = function (name, data) {
-        slackHandler.sendMessage(slackHandler.renderMessage({ type: name, data: data }));
+        //console.log(JSON.stringify(data));
```
Comment (tobilg) [inline index.js]: I'd like to encapsulate the filtering in the `lib/SlackHandler.js` file, not in the main `index.js`

File: index.js [hunk 4]
```diff
@@ -33,16 +34,53 @@ if (process.env.EVENT_TYPES) {
...
+        if (options.whitelistRegEx.length > 0) {
+          var events = filterEventsByAppId(data);
+          events.forEach(function(ev) { sendSlackMessage(name, ev); });          
+        } else {
+          sendSlackMessage(name, data);           
+        }
     }
 });
 
+function filterEventsByAppId(data){
```
Comment (tobilg) [inline index.js]: Please move this to `lib/SlackHandler.js`

File: index.js [hunk 5]
```diff
@@ -33,16 +34,53 @@ if (process.env.EVENT_TYPES) {
...
+    if (data.currentStep && data.currentStep.actions){
+      data.currentStep.actions.forEach(function(action){
+        if (rx.test(action.app.toLowerCase())) events.push(data); 
+      });
+    }
+  });
+  return events;
+}
+
+function sendSlackMessage(name, data){
```
Comment (tobilg) [inline index.js]: Is this single-line function even necessary?

**Review by tobilg — CHANGES_REQUESTED** [review]

**Comment by tobilg** [comment]
Do you also need an update of the DC/OS Universe package @eodgooch?

File: index.js (hunk 2 above)
Comment (eodgooch) [reply index.js]: haha of course

File: index.js (hunk 5 above)
Comment (eodgooch) [reply index.js]: not necessary but my standard practice is to write a function if I'm writing the same line of code 2+ times.

File: index.js (hunk 1 above)
Comment (eodgooch) [reply index.js]: that's a great insight. I completely overlooked it. The intent was to allow a comma separated list of regex similar to your `EVENT_TYPES`.

**Comment by eodgooch** [comment]
I'll go ahead and update the DC/OS package once you approve updates. What version tag? `0.3.2`?

I am also considering adding an env var to log events. Not sure if this is something you want in the project. My goal is to log events to get picked up by our shipper and sent to ES. Let me know.

**Comment by tobilg** [comment]
I implemented the desired changes in https://github.com/tobilg/marathon-slack/commit/af88c18c77ed0563b90b660750df986c03b310fa The Docker image is currently building in version `0.4.0`.

**Pull request closed without merging**